src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1636,11 +1637,43 @@ private bool isStructRequiringStackAllocRetBuf(CORINFO_CLASS_STRUCT_* cls)
         }
 
         private CORINFO_MODULE_STRUCT_* getClassModule(CORINFO_CLASS_STRUCT_* cls)
-        { throw new NotImplementedException("getClassModule"); }
+        {
+            TypeDesc type = HandleToObject(cls).GetTypeDefinition();
+            foreach (MethodDesc method in type.GetAllMethods())
+            {
+                MethodIL methodIL = _compilation.GetMethodIL(method);
+                if (methodIL != null)
+                {
+                    return ObjectToHandle(methodIL);
+                }
+            }
+            return null;
+        }
+
         private CORINFO_ASSEMBLY_STRUCT_* getModuleAssembly(CORINFO_MODULE_STRUCT_* mod)
-        { throw new NotImplementedException("getModuleAssembly"); }
+        {
+            MethodIL methodIL = HandleToObject(mod);
+            EcmaMethod method = methodIL.OwningMethod.GetTypicalMethodDefinition() as EcmaMethod;
+            IAssemblyDesc assembly = method?.Module.Assembly;
+
+            return (CORINFO_ASSEMBLY_STRUCT_*)ObjectToHandle(assembly);
+        }
+
+        private Dictionary<string, IntPtr> _assemblyNameCache = new Dictionary<string, IntPtr>();
+        private UTF8Encoding _assemblyNameEncoding = new UTF8Encoding();
+
```

Comment:
We do not have caches like in any other similar places. What makes this place special to need this cache?

I think this should use the same pattern as e.g. `getMethodName`. If it is useful to cache the pinned byte arrays (unclear whether it is a win), it should be in the centrally in the `GetPin` method.

Reply:
I have removed the cache in 2nd commit per your suggestion. I must admit I'm still not completely clear on the expected behavior though - at the first glance it seems like repeated calls to <code>getAssemblyName</code> end up repeatedly calling into GetPin and potentially / ultimately leaking the unmanaged heap as the JIT interface contract doesn't seem to provide any means to un-pin (and ultimately GC collect) a previously pinned assembly name.

Reply:
We assume that this can be only called so many times when compiling given method. We clean this up in bulk at the end of each method compilation (look for `CompileMethodCleanup`).

Reply:
Sounds good, thanks for clarifying!